atom/common/promise_util.h

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2018 GitHub, Inc.
+// Use of this source code is governed by the MIT license that can be
+// found in the LICENSE file.
+
+#ifndef ATOM_COMMON_PROMISE_UTIL_H_
+#define ATOM_COMMON_PROMISE_UTIL_H_
+
+#include <string>
+
+#include "native_mate/converter.h"
+
+namespace atom {
+
+namespace util {
+
+class Promise {
+ public:
+  explicit Promise(v8::Isolate* isolate) {
+    isolate_ = isolate;
+    resolver_.Reset(isolate, v8::Promise::Resolver::New(isolate));
+  }
+  ~Promise() { resolver_.Reset(); }
```

Comment:
Slightly confused by this -- it looks like v8::Global::~v8::Global() calls reset() on its own?

```cpp
  V8_INLINE ~Global() { this->Reset(); }
```

Reply:
I think this is left over from when I was fighting compilation warning things, will remove